api/client/proto/authservice.proto

```diff
@@ -780,12 +786,48 @@ message DeleteMFADeviceRequestInit {
 // DeleteMFADeviceResponseAck is a confirmation of successful device deletion.
 message DeleteMFADeviceResponseAck {}
 
-// GetMFADeviceRequest is a request for MFA devices for the calling user.
-message GetMFADevicesRequest {}
+// GetMFADevicesRequest is a request for MFA devices for the calling user.
+message GetMFADevicesRequest {
+    // Request defines how the user will be verified before receiving list of devices.
+    oneof Request {
+        // TokenID verifies existence of token and retrieves devices for user defined in token.
+        string TokenID = 1;
+        // AuthenticatedUser is a user who has beeng authenticated (logged in) and can
+        // view their own devices.
+        string AuthenticatedUser = 2;
+    }
+}
 
 // GetMFADeviceResponse is a response for GetMFADevices RPC.
 message GetMFADevicesResponse { repeated types.MFADevice Devices = 1; }
 
+// DeleteMFADeviceNonstreamRequest is a request to delete a mfa device for the user defined in
+// token.
+message DeleteMFADeviceNonstreamRequest {
+    // TokenID is the ID of a user token.
+    string TokenID = 1;
```

Comment:
This should be an MFAAuthenticateResponse - it should be able to call with either token or a signed challenge.